src/VirtoCommerce.CatalogModule.Core/Model/Search/CatalogSearchCriteria.cs

```diff
@@ -1,9 +1,14 @@
+using System.Linq;
 using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.CatalogModule.Core.Model.Search
 {
     public class CatalogSearchCriteria : SearchCriteriaBase
     {
-        public string[] CatalogIds { get; set; }
+        public string[] CatalogIds
+        {
+            get => ObjectIds.ToArray();
```

Comment:
Is NullreferenceException possible here?

Reply:
fixed